src/Couchbase.Lite.Shared/Query/IndexConfiguration.cs

```diff
@@ -28,53 +31,52 @@ namespace Couchbase.Lite.Internal.Query
     /// </summary>
     public abstract class IndexConfiguration
     {
-        #region Properties
+        private const string Tag = nameof(IndexConfiguration);
 
         /// <summary>
         /// Gets the expressions to use to create the index
         /// </summary>
         public string[]? Expressions { get; }
 
-        internal C4QueryLanguage QueryLanguage { get; }
-
         internal C4IndexType IndexType { get; }
 
         internal abstract C4IndexOptions Options { get; }
 
-        #endregion
-
-        #region Constructor
-
-        internal IndexConfiguration(C4IndexType indexType, params string[] items)
-            : this(indexType, C4QueryLanguage.N1QLQuery)
-        {
-            Expressions = items;
-        }
-
-        internal IndexConfiguration(C4IndexType indexType, C4QueryLanguage queryLanguage)
+        internal IndexConfiguration(C4IndexType indexType, string[]? expressions)
         {
+            if(indexType == C4IndexType.ArrayIndex && expressions != null) {
+                // Quick sanity check.  If I allow this to proceed, no error will happen but
```

Comment:
There was an extended discussion with Jens, about what, exactly, we should check, here.  His position was that we should leave as much as possible to LiteCore; that the platforms should do as little checking as possible.
The conclusion was that the platform APIs should follow these rules:
- If the client code passes `null` as the expression list, we should pass an empty string to LiteCore as the expression spec
- It is an error for the client code to pass a singleton list containing null or the empty string
- All other espression specs should be passed, comma separated, to LiteCore, for checking.  In particular, an empty list should be legal (it is a synonym for passing null).

Reply:
Okey dokey guess I'll be silently changing the behavior for 4.0 then.